docs/source/paper.md

```diff
@@ -0,0 +1,102 @@
+---
+title: 'kiwiPy: Robust, high-volume, messaging for big-data and computational science workflows'
+tags:
+  - python
+  - high-throughput computing
+  - message broker
+  - scientific workflows
+  - fault-tolerant computing
+authors:
+  - name: Martin Uhrin
+    orcid: 0000-0001-6902-1289
+    affiliation: "1, 2, 3"
+  - name: Sebastiaan P. Huber
+    orchid: 0000-0001-5845-8880
+    affiliation: "2, 3"
+affiliations:
+ - name: Department of Energy Conversion and Storage, Technical University of Denmark, 2800 Kgs. Lyngby, Denmark
+   index: 1
+ - name: National Centre for Computational Design and Discovery of Novel Materials (MARVEL), École Polytechnique Fédérale de Lausanne, CH-1015 Lausanne, Switzerland
+   index: 2
+ - name: Theory and Simulation of Materials (THEOS), Faculté des Sciences et Techniques de l’Ingénieur, École Polytechnique Fédérale de Lausanne, CH-1015 Lausanne, Switzerland
+   index: 3
+date: 19 April 2020
+bibliography: paper.bib
+---
+
+# Summary
+
+The computational sciences have seen a huge increase in the use of high-throughput, automated, workflows over the course 
+of the last two decades or so.  Focusing on just our domain of computational materials science there have
+been several large scale initiatives to provide high-quaity results from standardised calculations 
+[@Landis2012; @Curtarolo2012; @Jain2013; @Saal2013; @Draxl2019a; @Talirz2020].  Almost all of these repositories are
+populated using results from high-throughput quantum mechanical calculations that rely on workflow frameworks 
+[@Jain2015a; @Mortensen2020], including our own [AiiDA](http://www.aiida.net/) [@Pizzi2016; @Huber2020] which powers the 
+[Materials Cloud](https://www.materialscloud.org/).
+One of the many challenges for such frameworks is maximising fault-tolerance whilst simultaneously maintaining high-throughput, 
+often across several systems (typically the client launching the tasks, the supercomputer carrying out the computations
+and the server hosting the database).  
+
+On the software level these problems are perhaps best addressed by using messaging
+brokers that take responsibility for guaranteeing the durability and atomicity of messages and often enable event based
+communication.  Indeed solutions such as [RabbitMQ](https://www.rabbitmq.com/) see widespread adoption in industry, however adoption
+in academia has been more limited, with home-made queue data structure, race condition susceptible locks and polling based
+solutions common. This is likely due to the relatively complex nature of some of the APIs (which
```

Comment:
```suggestion
solutions being commonplace. This is likely due to the typically complex APIs (which
```

Reply:
'being commonplace' accepted.

API's bit changed to:
'This is likely due to message brokers typically having complex APIs'